Detectors/TPC/workflow/src/CalibdEdxSpec.cxx

```diff
@@ -51,34 +56,35 @@ class CalibdEdxDevice : public Task
     const auto fitSnp = ic.options().get<bool>("fit-snp");
 
     mDumpToFile = ic.options().get<bool>("file-dump");
-    auto field = ic.options().get<float>("field");
-
-    if (field <= -10.f) {
-      const auto inputGRP = o2::base::NameConf::getGRPFileName();
-      const auto grp = o2::parameters::GRPObject::loadFrom(inputGRP);
-      if (grp != nullptr) {
-        field = 5.00668f * grp->getL3Current() / 30000.;
-        LOGP(info, "Using GRP file to set the magnetic field to {} kG", field);
-      }
-    }
 
     mCalib = std::make_unique<CalibdEdx>(dEdxBins, mindEdx, maxdEdx, angularBins, fitSnp);
     mCalib->setApplyCuts(false);
     mCalib->setSectorFitThreshold(minEntriesSector);
     mCalib->set1DFitThreshold(minEntries1D);
     mCalib->set2DFitThreshold(minEntries2D);
-    mCalib->setField(field);
     mCalib->setElectronCut(fitThreshold, fitPasses);
   }
 
+  void finaliseCCDB(o2::framework::ConcreteDataMatcher& matcher, void* obj) final
+  {
+    o2::base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj);
+  }
+
   void run(ProcessingContext& pc) final
   {
+    o2::base::GRPGeomHelper::instance().checkUpdates(pc);
     const auto tfcounter = o2::header::get<DataProcessingHeader*>(pc.inputs().get("tracks").header)->startTime;
     const auto tracks = pc.inputs().get<gsl::span<TrackTPC>>("tracks");
 
     LOGP(info, "Processing TF {} with {} tracks", tfcounter, tracks.size());
-    mRunNumber = processing_helpers::getRunNumber(pc);
     mCalib->fill(tracks);
+
+    // store run number and CCDB time only once
+    if (mTimeStampStart == 0 || pc.services().get<o2::framework::TimingInfo>().timeslice) {
```

Comment:
sorry, here there should be `if (mTimeStampStart == 0 || pc.services().get<o2::framework::TimingInfo>().timeslice==0) {`
This still may change as the framework will provide a method to detect run start after stop/start procedure.